enterprise/server/raft/filestore/filestore.go

```diff
@@ -35,7 +34,111 @@ type WriteCloserMetadata interface {
 	MetadataWriter
 }
 
-func NewWriter(ctx context.Context, fileDir string, wb pebble.Writer, fileRecord *rfpb.FileRecord) (WriteCloserMetadata, error) {
+// returns partitionID, isolation, hash or
```

Comment:
could use a a better comment. should make it clear that we always return the same elements but one of them can be empty depending on isolation type.

Reply:
Done.